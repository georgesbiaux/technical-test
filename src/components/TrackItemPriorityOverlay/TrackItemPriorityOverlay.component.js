import React from 'react';
import StyledTrackItemPriorityOverlay from './TrackItemPriorityOverlay.style';

const TrackItemPriorityOverlay = () => (
  <StyledTrackItemPriorityOverlay className="track-priority-overlay">
    <div className="priority-icon-container">
      <i className="priority-icon fa fa-share" />
    </div>
    <div className="bottom-border" />
  </StyledTrackItemPriorityOverlay>
);

export default TrackItemPriorityOverlay;
