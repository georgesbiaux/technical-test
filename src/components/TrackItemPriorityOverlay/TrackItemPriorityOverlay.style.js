import styled from 'styled-components';

const StyledTrackItemPriorityOverlay = styled.div`
  .priority-icon-container {
    border-color: transparent transparent transparent #75d286;
    position: absolute;
    z-index: 2;
    bottom: 0;
    left: 0;
    width: 0;
    height: 0;
    border-style: solid;
    border-width: 30px 0 0 30px;
  }

  .priority-icon {
    position: absolute;
    z-index: 3;
    bottom: 4px;
    left: -28px;
    transform: rotate(-90deg);
    font-size: 14px;
    color: #ffffff;
  }

  .bottom-border {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 0;
    height: 5px;
    background-color: #75d286;
    z-index: 1;
  }
`;

export default StyledTrackItemPriorityOverlay;
