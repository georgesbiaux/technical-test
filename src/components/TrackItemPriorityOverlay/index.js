export { default } from './TrackItemPriorityOverlay.component';
