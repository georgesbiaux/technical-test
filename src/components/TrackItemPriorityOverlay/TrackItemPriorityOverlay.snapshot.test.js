import React from 'react';
import { render } from 'enzyme';
import toJSON from 'enzyme-to-json';
import TrackItemPriorityOverlay from './TrackItemPriorityOverlay.component';

describe('<TrackItemPriorityOverlay>', () => {
  it('should render correctly', () => {
    const tree = render(<TrackItemPriorityOverlay />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
