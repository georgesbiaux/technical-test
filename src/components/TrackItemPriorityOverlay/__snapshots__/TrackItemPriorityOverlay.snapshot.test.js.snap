// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackItemPriorityOverlay> should render correctly 1`] = `
<div
  class="track-priority-overlay sc-bdVaJa bmYYPN"
>
  <div
    class="priority-icon-container"
  >
    <i
      class="priority-icon fa fa-share"
    >
      
    </i>
  </div>
  <div
    class="bottom-border"
  >
    
  </div>
</div>
`;
